File: server/ruby/app/infrastructure/mappers/payment_details/amount_of_money_mapper.rb
module Infrastructure
  module Mappers
    module PaymentDetails
      class AmountOfMoneyMapper
        def self.from_sdk_response(response)
          return nil if response.nil?

          dto = Business::Domain::Payments::PaymentDetails::AmountOfMoney.new
          dto.amount = response.amount
          dto.currency_code = response.currency_code

          dto
        end
      end
    end
  end
end

File: server/python/app/infrastructure/mappers/payment_details/amount_of_money_mapper.py
from typing import Optional
from app.application.domain.payments.payment_details.amount_of_money import AmountOfMoney
from onlinepayments.sdk.domain.amount_of_money import AmountOfMoney as AmountOfMoneySdk

class AmountOfMoneyMapper:

    @staticmethod
    def map_from_sdk_response(response: Optional[AmountOfMoneySdk]) -> Optional[AmountOfMoney]:
        if response is None:
            return None

        dto = AmountOfMoney()
        dto.amount = response.amount
        dto.currency_code = response.currency_code

        return dto